fv3core/fv3core/stencils/fxadv.py

```diff
@@ -36,6 +42,11 @@ def main_uc_contra(
         ):
             uc_contra = utmp
 
+        with horizontal(region[:, local_js - 1 : local_je + 3]):
+            # for C-grid, u must be regridded to lie at same point as v
+            u = 0.25 * (uc[0, -1, 0] + uc[1, -1, 0] + uc + uc[1, 0, 0])
+            vc_contra = contravariant(vc, u, cosa_v, rsin_v)
```

Comment:
How come you don't need to keep this for v but you need it for u above:

```        
with horizontal(region[:, j_start], region[:, j_end + 1]):
    vc_contra = vtmp
```

Reply:
That's a good question I only have a partial answer for. Removing this for this block only changes values in the outermost halo, while removing this for `u` also modifies eight points in the second ring of halos. Why this is, I can't say. Because we depend computationally on other points in the second ring of halos, it's more complicated to enable selective validation for that case.